src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -479,16 +477,18 @@ private static void enableFirestore(
   static String extractOperationIdFromName(final String appId, final String opName) {
     // The format returns is apps/{appId}/operations/{useful id} so we need to extract it
     // Add a check in case they ever change the format
-    final String uuidRegex =
-        "\\p{XDigit}{8}-\\p{XDigit}{4}-\\p{XDigit}{4}-\\p{XDigit}{4}-\\p{XDigit}{12}";
-    final Pattern pattern =
-        Pattern.compile(String.format("^apps/%s/operations/(%s)", appId, uuidRegex));
-    final Matcher matcher = pattern.matcher(opName);
-    if (!matcher.find()) {
+    String expectedPrefix = String.format("apps/%s/operations/", appId);
+    if (opName.startsWith(expectedPrefix)) {
+      var split = opName.split("/");
+      if (split.length == 4) {
+        return split[3];
+      }
```

Comment:
Given that you're already matching the prefix, checking for the split count result seems like duplication of effort. Instead you could write
```
      return opName.substring(opName.lastIndexOf("/") + 1);
```
or
```
      return opName.substring(expectedPrefix.length());
```